### ISSUE TITLE
undefined method `recreate_digest_image_versions' for #<Mailing:0x007fc2fe2ab548>

### ISSUE THREAD

**Issue by sekrett** [kirtithorat/carrierwave-crop#12]
I looked the source code, it should be handled with `method_missing` but the `missing_method` is never called, I checked with debugger. I have Rails 4.2.1.

**Comment by kirtithorat** [comment]
@sekrett Could you please provide the stack trace of the error and the steps to reproduce it.
If possible, extract the minimum possible code that gives the error in a repo and share it.

**Comment by sekrett** [comment]
It may be difficult to find, who is overriding the `missing_method` in a big project with many gems.
I think here it is not necessary to use meta programming at all. Instead of:

<pre>after_update :"recreate_#{attachment}_versions"</pre>

you could try:

<pre>after_update do
  recreate_cropped_versions(attachment)
end
....
def recreate_cropped_versions(attachment)
end
</pre>

Makes sense? I can send a pull request to fix this if you wish.

**Comment by kirtithorat** [comment]
@sekrett As per my understanding, the code should work as it is with `"recreate_#{attachment}_versions"`. Here `attachment` refers to the image field. For example:

```
class User < ActiveRecord::Base
  mount_uploader :avatar, AvatarUploader
  crop_uploaded :avatar  ## Add this
  ## ...
end
```

In the above code `attachment` would refer to `avatar`, so you will get a callback method named `recreate_avatar_versions` .  My guess would be that you have messed up the `carrierwave-crop` configuration so just make sure your code is along the same lines as specified above. I asked for stack trace/ example repo just to confirm. Anyways try the code, in the above specified manner and it should work. Let me know if you still face problem.

**Comment by sekrett** [comment]
I have read the documentation and checked everything 3 times.

Of course the code should work but it does not in my app. As I said `method_missing` is not called, but the `recreate_avatar_versions` IS called, so the error comes out. I have added `crop_uploaded`, in fact it is WHO tries but fails to call the `missing_method`.

I'm really in doubt that using `missing_method` is absolutely necessary in your gem. `missing_method` is useful when you don't know which method will be called, but here you know it. If it is possible I would prefer to patch your gem to do direct calls without `missing_method`, which will always work. Maybe this will work:

<pre>
recreate_cropped_versions("#{attachment}")
</pre>

or the code I wrote before. I have to test if attachment is a dynamic variable or should be passed as a string, just need some time to play with it.

Actually your plugin is great. There is no bug. Simply I offer you a better implementation which has more chances to work in different apps.
